### PR TITLE
Student rep maintenance dn

### PR DESCRIPTION
**Pull request by dbnorth**
Update to support student rep maint for including piece, instrument and semester data and ordering by semester and instrument.

### CODE REVIEW

File: app/controllers/eventSignup.controller.js [hunk 1]
```diff
@@ -93,6 +102,23 @@ exports.findByEventId = (req, res) => {
     where: {
       eventId: id,
     },
+    include: {
+      model: db.studentInstrumentSignup,
+      required: false,
```
Comment (AndrewClubb) [inline app/controllers/eventSignup.controller.js]: Why is this false?

File: app/controllers/studentPiece.controller.js [hunk 2]
```diff
@@ -159,25 +159,45 @@ exports.deleteAll = (req, res) => {
...
+        model: db.piece,
+        required: true,
+        include: { model: db.composer, required: true },
+      },
+      { model: db.semester, required: true },
+      {
+        model: db.piece,
+        required: true,
+        include: { model: db.composer, required: true },
+      },
```
Comment (AndrewClubb) [inline app/controllers/studentPiece.controller.js]: including semester and piece is duplicated

File: app/controllers/studentPiece.controller.js [hunk 3]
```diff
@@ -159,25 +159,45 @@ exports.deleteAll = (req, res) => {
...
+      {
+        model: db.studentInstrument,
+        required: true,
+
+        include: [
+          { model: db.instrument, required: true },
+          { model: db.instrument, required: true },
```
Comment (AndrewClubb) [inline app/controllers/studentPiece.controller.js]: include instrument is duplicated

File: app/controllers/studentPiece.controller.js [hunk 4]
```diff
@@ -159,25 +159,45 @@ exports.deleteAll = (req, res) => {
...
-        },
-        { model: db.instrument, required: true },
-      ],
-    },
+        ],
+      },
+    ],
+    order: [
+      [db.semester, "name", "asc"],
+      [db.studentInstrument, db.instrument, "name", "asc"],
```
Comment (AndrewClubb) [inline app/controllers/studentPiece.controller.js:195]: Do we want to hard code these, or pass these orders by parameter?

**Review by AndrewClubb — DISMISSED** [review]

File: app/controllers/eventSignup.controller.js (hunk 1 above)
Comment (dbnorth) [reply app/controllers/eventSignup.controller.js]: don't know, fixed.  It seems like there would alwasy be one.

File: app/controllers/studentPiece.controller.js (hunk 4 above)
Comment (dbnorth) [reply app/controllers/studentPiece.controller.js:195]: don't want to fix it now.

File: app/controllers/studentPiece.controller.js (hunk 3 above)
Comment (dbnorth) [reply app/controllers/studentPiece.controller.js]: fixed

**Review by JuliaFlores21 — APPROVED** [review]
looks good

**Pull request merged**